Prerelease adjustments (#34)
rsksmart/rif-communications-pubsub-node: README.md

![](https://img.shields.io/badge/npm-%3E%3D6.0.0-orange.svg?style=flat-square)
![](https://img.shields.io/badge/Node.js-%3E%3D10.0.0-orange.svg?style=flat-square)

# RIF Communications bootnode

> ## Getting started:

Reviewer: ```suggestion
## Getting started
```

(no need for `:` in titles)

> - Node 14+ (14.13 recommended)

Reviewer: ```suggestion
### Prerequisites
- Node 14+ (14.13 recommended)
```

### Get the code

> Download the code by executing `git clone https://github.com/rsksmart/rif-communications-pubsub-bootnode -b grpc-api`.

Reviewer: i would use ``` for this command and enter it in a new line

After this, run `npm install` inside the cloned folder.

### Create a key for the RIF Communications bootnode

> 1. Create the `config/keys/lumino` folder.

Reviewer: ```suggestion
1. Create the `config/keys/client` folder inside the cloned directory.
```

> 2. Execute the following commands inside this folder:

Reviewer: ```suggestion
2. Execute the following commands inside this new folder:
```

```
openssl ecparam -genkey -name secp256k1 -out ecs_key.pem -param_enc explicit
``` 
and then

```
openssl pkcs8 -in ec_key.pem -topk8 -v2 aes-256-cbc -v2prf hmacWithSHA256 -outform DER -out ec_key_pkcs8_v2.der
``` 
The latter one will require you to define a password.

### Edit the private key configuration

- go back to `config` folder

> - edit the `lumino.json5` file. Modify the `key.password` value to your key's password

Reviewer: despite PR #33, i think we should probably change this. we shouldn't need to make a direct reference to `lumino` in these configs.

i propose we keep the `lumino.json5` around, but also add another base config (`client.json5`) without the bootnotes, and then make a reference to it in these sections.

Author: Agree. I'd keep lumino.json5 but also add client.json5. I'll change all the references to lumino.json5 to client.json5. Agree?

> - change the `key.privateKeyURLPath` to the config key path `[...]/config/keys/lumino/ec_key_pkcs8_v2.der`

Reviewer: ```suggestion
1. Go back to the `config` folder
2. Edit the `client.json5` file:
    1. Modify the `key.password` value to your key's password
    2. Change the `key.privateKeyURLPath` to the config key path `[...]/config/keys/client/ec_key_pkcs8_v2.der`
```

### Configure the Bootstrap nodes

> In order to access the network, you need to connect to at least one node 

Reviewer: ```suggestion
In order to access a network, you need to connect to at least one node 
```

> that's are already in it. You can configure them using the the following parameters:

Reviewer: ```suggestion
that's are already in it. You can configure these using the the following parameters:
```

```json5
{
  libp2p: {
    config: {
      peerDiscovery: {
        bootstrap: {

>           // Enable bootstraping

Reviewer: ```suggestion
          // Enable bootstrapping
```

          enabled: true,
          // list of nodes to connect by default

>           // e.g. Lumino Networkd bootstrap nodes

Reviewer: ```suggestion
          // e.g. Lumino Network bootstrap nodes
```

          list: [
            "/ip4/18.214.23.85/tcp/5011/p2p/16Uiu2HAmAxP26UzDG3drx1ikopjMYK6Zseyud9qJVoshZ5RgTowJ",
            "/ip4/3.228.1.178/tcp/5011/p2p/16Uiu2HAm9Z9zSbXHHtSnjk2iCjnmBcb2ZXSA694jLCwAUUatqmGq",
            "/ip4/18.206.56.242/tcp/5011/p2p/16Uiu2HAmRzgNWzwMivPCLRvLadbmLGPrymV8rxtBeq7PhndidQ6h"
          ],
        }
      }
    }
  }
}
```
### Start the node

- At root folder, run `NODE_ENV=lumino npm run api-server`

> - You should see a log output similar to this one:

Reviewer: ```suggestion
At the project root folder, run `NODE_ENV=client npm run api-server`

You should see a log output similar to this one:
```

```
[INFO] 11:34:08 ts-node-dev ver. 1.1.1 (using ts-node ver. 9.1.1, typescript ver. 4.1.3)
Loading encrypted DER key
Node started, listening on addresses:
/ip4/127.0.0.1/tcp/5011/p2p/16Uiu2HAmQswXYkmzDTgs2Em5JkhP8Y33CEhXQUHY3trctB1StTe7
/ip4/127.0.0.1/tcp/5012/ws/p2p/16Uiu2HAmQswXYkmzDTgs2Em5JkhP8Y33CEhXQUHY3trctB1StTe7

Listening on topics: 

PEERID: 16Uiu2HAmQswXYkmzDTgs2Em5JkhP8Y33CEhXQUHY3trctB1StTe7
GRPC Server started on port 5013
```

## Additional settings

> ### Exposing RIF Communications from a private IP address

Reviewer 2: 👏🏻
